ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ModulesInstaller.java

```diff
@@ -207,32 +204,19 @@ private void doInstallMissingModules () throws OperationException {
         if (! installContainer.listInvalid ().isEmpty ()) {
             throw new IllegalArgumentException ("Some are invalid for install: " + installContainer.listInvalid ());
         }
-        InstallSupport installSupport = installContainer.getSupport ();
-        if (downloadHandle == null) {
-            downloadHandle = ProgressHandleFactory.createHandle (
-                getBundle ("InstallerMissingModules_Download",
-                presentUpdateElements (finder.getVisibleUpdateElements (modules4install))));
-        }
-        progressMonitor.onDownload(downloadHandle);
-        Validator v = installSupport.doDownload (downloadHandle, false);
-        if (verifyHandle == null) {
-            verifyHandle = ProgressHandleFactory.createHandle (
-                    getBundle ("InstallerMissingModules_Verify"));
-            }
-        progressMonitor.onValidate(verifyHandle);
-        Installer i = installSupport.doValidate (v, verifyHandle);
-        if (installHandle == null) {
-            installHandle = ProgressHandleFactory.createHandle (
-                    getBundle ("InstallerMissingModules_Install"));
+        boolean ok = PluginManager.openInstallWizard(installContainer);
+        if (!ok) {
+            StringBuilder sb = new StringBuilder();
```

Comment:
For what it's worth, you might want to use a Stream instead (with a static import for `Collectors::joining`):

```suggestion
            String s = modules4install.stream()
                .map(UpdateElement::getDisplayName)
                .collect(joining(", "));
```